### ISSUE TITLE
Conditional application based on user consent

### ISSUE THREAD

**Issue by chillu** [sminnee/silverstripe-tagmanager#17]
User consent is probably out of scope for this module, but the reality is that most of the scripts inserted through tag managers have privacy implications. And this kind of tracking is increasingly seen as a technology which requires explicit consent by users. The original version of this was the "EU cookie law", and now it's much more granular/explicit with GDPR. Since pretty much all websites have to reasonably assume that they're dealing with EU citizens, I think we can't just provide tag management insertion with auto-tracking (such as Google Analytics). There need to be some signals around that insertion (e.g. conditionals on cookies).

In other words: In the current state of the module, we enable web administrators to break privacy regulations by default, and expose the companies they work for to legal peril. Devs are just as likely to make those mistakes without a tag manager, but there's usually a bit more "business analysis" involved in those implementations.

**Comment by chillu** [comment]
Wagtail's tag manager has a built in cookie bar: https://github.com/jberghoef/wagtail-tag-manager#wtm_cookie_bar and https://github.com/jberghoef/wagtail-tag-manager#concept

**Comment by sminnee** [comment]
In terms of architecture, the TagInserter is the thing that's currently responsible for this. Right now it's a fairly broad service – it's simply a controller extension, and has a mixture of responsibilities.

We could probably refactor TagInserter out to have a couple of more precisely-defined services, e.g.

 * SnippetListProvider - service to return the snippets for the current pageview
 * SnippetWriter - service to update a body/httpresponse with the content of the given snippets

We could then have 1 of these able to be swapped out with a service that blocks snippets with an appropriate cookie bar.

The cookie/tracking bar itself is probably broader in scope than the tagmanager as it would cover our use of session cookies too. Maybe it's still bundled with this module for simplicity, or it might be a separate module that interacts with this. The latter would make sense if there's already a cookiebar module we want to make use of.

As an aside, cookie bars are horrible and it would be great to consider how we can ensure that they are less so.
